lib/opentelemetry-bridge/OTelTracer.js

```diff
@@ -0,0 +1,147 @@
+    // We want to get a core Transaction or Span as a parent, when possible,
+    // because that is required to support the span compression feature.
+    let parentGenericSpan
+    let parentOTelSpanContext
+    if (otelSpanOptions.root) {
+      // Pass through: explicitly want no parent.
+    } else if (otelContext instanceof OTelBridgeRunContext) {
+      parentGenericSpan = otelContext.currSpan() || otelContext.currTransaction()
+      if (parentGenericSpan instanceof OTelBridgeNonRecordingSpan) {
+        // This isn't a real Transaction we can use. It is a placeholder
+        // to propagate its SpanContext. Grab just that.
+        parentOTelSpanContext = parentGenericSpan.spanContext()
+        parentGenericSpan = null
+      }
+    } else {
+      // `otelContext` is any object that is meant to satisfy `interface
+      // Context`. This may hold an OTel `SpanContext` that should be
+      // propagated.
+      parentOTelSpanContext = otel.trace.getSpanContext(otelContext)
+    }
+
+    // Create the new Span/Transaction.
+    let newTransOrSpan = null
+    if (parentGenericSpan) {
+      // New child span.
+      const trans = parentGenericSpan instanceof Transaction ? parentGenericSpan : parentGenericSpan.transaction
+      const spanOpts = {
+        childOf: parentGenericSpan
+      }
+      if (otelSpanOptions.startTime) {
+        spanOpts.startTime = epochMsFromOTelTimeInput(otelSpanOptions.startTime)
+      }
+      newTransOrSpan = trans.createSpan(name, spanOpts)
+
+      // There might be no span, e.g. if transactionMaxSpans was hit. We have
+      // to return some OTelSpan, and we also want to propagate the parent's
+      // trace-context, if any.
+      if (!newTransOrSpan) {
+        return otel.trace.wrapSpanContext(otelSpanContextFromTraceContext(parentGenericSpan._context))
+      }
+    } else if (parentOTelSpanContext && otel.isSpanContextValid(parentOTelSpanContext)) {
+      // New continuing transaction.
+      // Note: This is *not* using `SpanContext.isRemote`. I am not sure if it
+      // is relevant unless using something, like @opentelemetry/core's
+      // `W3CTraceContextPropagator`, that sets `isRemote = true`. Nothing in
+      // @opentelemetry/api itself sets isRemote.
+      const transOpts = {
+        childOf: traceparentStrFromOTelSpanContext(parentOTelSpanContext)
+      }
+      if (parentOTelSpanContext.traceState) {
+        transOpts.tracestate = parentOTelSpanContext.traceState.serialize()
+      }
+      if (otelSpanOptions.startTime !== undefined) {
+        transOpts.startTime = epochMsFromOTelTimeInput(otelSpanOptions.startTime)
+      }
+      newTransOrSpan = this._ins.createTransaction(name, transOpts)
+    } else {
+      // New root transaction.
+      const transOpts = otelSpanOptions.startTime === undefined
+        ? {}
+        : { startTime: epochMsFromOTelTimeInput(otelSpanOptions.startTime) }
+      newTransOrSpan = this._ins.createTransaction(name, transOpts)
```

Comment:
That is my understanding of the OTel tracing spec: https://github.com/elastic/apm/blob/main/specs/agents/tracing-api-otel.md#compatibility-mapping

There is an inference algorithm for setting `transaction.type`. It defaults to 'unknown' (rather than our API's effective default of 'custom' for `apm.startTransaction(name)`) unless there are some particular span kind and attribute values.